core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManager.java

```diff
@@ -20,5 +28,87 @@
 
    void checkBulkRead();
 
+   @Deprecated
+      //test use only. it can be removed if we update the tests
    CacheTopology getLastStableTopology();
+
+   /**
+    * Adds a partially aborted transaction.
+    * <p/>
+    * The transaction should be registered when it is not sure if the abort happens successfully in all the affected
+    * nodes. It can rethrow the exception is the transaction is not registered.
+    *
+    * @param globalTransaction the global transaction.
+    * @param affectedNodes     the nodes involved in the transaction and they must abort the transaction.
+    * @param lockedKeys        the keys locally locked.
+    * @param suspectException  the suspected exception. It contains the node who left the view.
+    */
+   void addPartialRollbackTransaction(GlobalTransaction globalTransaction, Collection<Address> affectedNodes,
+                                      Collection<Object> lockedKeys, SuspectException suspectException);
```

Comment:
With the ISPN-4546 changes, I'd want the SuspectException (or OutdatedTopologyException, as it is now) to always be thrown, so that the StateTransferInterceptor can retry (and maybe the retry will throw an AvailabilityException).